tests: crypto: Add a unnormal testing condition

KangJianX opened the pull request:
For z_early_boot_rand_get(), give a length less than uint32 to be input
value, ensure the API can work normally with error condition to improve
code coverage.

Signed-off-by: Jian Kang <jianx.kang@intel.com>

enjiamai commented on tests/crypto/rand32/src/main.c:
```diff
@@ -30,15 +30,17 @@
 	z_early_boot_rand_get((uint8_t *)&last_gen, sizeof(last_gen));
 	z_early_boot_rand_get((uint8_t *)&gen, sizeof(gen));
-	zassert_true(last_gen != gen,
+	zassert_true(last_gen != gen && last_gen != tmp && tmp != gen,
```
Hi, does  "zassert_true(last_gen != gen && last_gen != tmp)" also get the same result we want? I guess check these two condition is enough.

KangJianX replied:
It cannot make sure if `gen` equal `tmp` by your comment.

enjiamai replied:
Got it! Thanks!

enjiamai reviewed (commented):
Could you please see my comment? to see if this can work or not. thanks!

LeiW000 commented on tests/crypto/rand32/src/main.c:
```diff
@@ -30,15 +30,17 @@
 	uint32_t buf[N_VALUES];
 
 	/* Test early boot random number generation function */
+	/* Add a condition about length less than uint32 */
```
It is better to change the comments like below.

" Cover the case, where argument "length" is < size of "size_t". "

KangJianX replied:
Done

KangJianX committed:
tests: crypto: Add a unnormal testing condition

For z_early_boot_rand_get(), give a length less than uint32 to be input
value, ensure the API can work normally with error condition to improve
code coverage.

Signed-off-by: Jian Kang <jianx.kang@intel.com>
